internal/awsservice/cloudwatchlogs.go

```diff
@@ -134,15 +73,27 @@ func DeleteLogGroup(logGroupName string) {
 	}
 }
 
-// ValidateLogsInOrder takes a log group, log stream, a list of specific log lines and a timestamp.
+// ValidateLogs takes a log group, log stream, a list of specific log lines and a timestamp.
 // It should query the given log stream for log events, and then confirm that the log lines that are
 // returned match the expected log lines. This also sanitizes the log lines from both the output and
 // the expected lines input to ensure that they don't diverge in JSON representation (" vs ')
-func ValidateLogsInOrder(t *testing.T, logGroup, logStream string, logLines []string, since time.Time) {
-	log.Printf("Checking %s/%s since %s for %d expected logs", logGroup, logStream, since.UTC().Format(time.RFC3339), len(logLines))
+func ValidateLogs(t *testing.T, logGroup, logStream string, since, until *time.Time, validator func(logs []string) bool) {
```

Comment:
This would be out of scope for the PR: would it be better to return error instead of utilizing `test` as a helper?

Reply:
We could, but what is the benefit of that? The purpose of this function is to validate logs from CloudWatch. We don't always need to wrap everything in an error and check it in the main test code. I don't see the issue with this.

Reply:
I ended up making the change in order to properly reuse it in the emf plugin test